desc/objectives/_coils.py

```diff
@@ -597,3 +597,208 @@ def compute(self, params, constants=None):
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: has no effect for this objective
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    field_grid : Grid, optional
+        Grid containing the nodes to evaluate field source at on
+        the winding surface. (used if e.g. field is a CoilSet or
+        FourierCurrentPotentialField). Defaults to the default for the
+        given field, see the docstring of the field object for the specific default.
+    eval_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate the normal magnetic field at
+        plasma geometry at. Defaults to a LinearGrid(L=eq.L_grid, M=eq.M_grid,
+        zeta=jnp.array(0.0), NFP=eq.NFP).
+    name : str, optional
+        Name of the objective function.
+    """
+
+    _coordinates = "rtz"
+    _units = "T*m^2"
+    _print_value_fmt = "Toroidal Flux: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        field,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        field_grid=None,
+        eval_grid=None,
+        name="toroidal-flux",
+    ):
+        if target is None and bounds is None:
+            target = eq.Psi
+        self._field = field
+        self._field_grid = field_grid
+        self._eval_grid = eval_grid
+        self._eq = eq
+
+        super().__init__(
+            things=[field],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self._eq
+        if self._eval_grid is None:
+            eval_grid = LinearGrid(
+                L=eq.L_grid, M=eq.M_grid, zeta=jnp.array(0.0), NFP=eq.NFP
+            )
+            self._eval_grid = eval_grid
+        eval_grid = self._eval_grid
+
+        errorif(
+            not np.allclose(eval_grid.nodes[:, 2], eval_grid.nodes[0, 2]),
+            ValueError,
+            "Evaluation grid should be at constant zeta",
+        )
+        if self._normalize:
+            self._normalization = eq.Psi
```

Comment:
use the `"Psi"` value from `compute_scaling_factors` instead

Reply:
It is the exact same `eq.Psi` though, so I did not think there was a need to calculate the rest of the scalings.

Reply:
Technically it introduces a tiny bit of overhead, but I would argue it's worth it to follow the same format as all of the other objective normalizations.